### ISSUE TITLE
panic: assignment to entry in nil map

### ISSUE THREAD

**Issue by michielbdejong** [cs3org/reva#3448]
This error messages seems to be happening on master since last week.

**Comment by michielbdejong** [comment]
Work-around: `git checkout 392b4` and `make build-revad`

**Comment by michielbdejong** [comment]
Confirmed if you `git checkout a2b2109` you get the panic.
Very weird that CI didn't pick this up when #3401 was reviewed!

**Comment by michielbdejong** [comment]
@glpatcern can you confirm?

**Comment by glpatcern** [comment]
Indeed weird that it was not picked up in the CI (and it didn't even fire doing tests). I'll have a look.

**Comment by glpatcern** [comment]
OK, issue understood and fixed, PR on the way

**Comment by michielbdejong** [comment]
You mean both the bug and the reason that CI didn't catch it?

**Comment by glpatcern** [comment]
The bug was due to a bad assumption in case of invalid config, about the CI I guess it does not test such invalid configs.

**Comment by michielbdejong** [comment]
Hm, I think we have two options:
* call an empty mimetypes config "invalid", in which case this should be documented on reva.link, and also the error message should be 'empty mimetypes config not allowed' instead of panic. This would require:
  * change the error message
  * add a test
  * add docs
  * change all production configs
* call an empy mimetypes config "valid", in which case no documentation change is needed, and no config change in production systems; we just need:
  * the bugfix you already wrote https://github.com/glpatcern/reva/commit/57d686cc3f88ca96b482d72e4eb373033cfe6a34
  * add a test where mimetype config is empty. This test would fail in a2b2109, but pass with your patch
 
I think the second one is less work?

**Comment by glpatcern** [comment]
Hi @michielbdejong, sorry I was too fast on Saturday: in fact a configuration with no parameters is perfectly valid for a driver such as the `demo` appprovider which, well, does not need parameters as it does not do much anyway.

So apart from doing "more or less work", the empty driver config is totally legitimate (and BTW that was the issue, not the empty mimetypes, which are fully optional on their own).

Now concerning a validation test, here we'd need to test for a demo driver with empty config. I can look into that.

**Comment by michielbdejong** [comment]
You can try loading [this config](https://github.com/cs3org/ocm-test-suite/blob/main/servers/revad/revad1.toml) in a test, I think you will be able to reproduce the panic with that config file and with your patch not applied.

**Comment by michielbdejong** [comment]
This issue was closed when the PR was merged, but we still need to write the test for it.

@glpatcern will you still write that test, or should I?

**Comment by glpatcern** [comment]
Yes, that was the intention but today I got stuck with other things (and I anticipate tomorrow will be the same...), will push a PR with the test on Wednesday.

**Comment by michielbdejong** [comment]
Is there a GitHub issue for the integration tests in the CI?
I'll close this now, thanks for all the digging you did!

**Comment by glpatcern** [comment]
It was a good exercise nevertheless! The GitHub issue is not yet there, we have an internal one (@gmgigi96 is working on it), but it will come soon.
